network/src/testutils/test_node.rs

```diff
@@ -345,3 +365,82 @@ pub fn mock_conn_metadata(
     }
     metadata
 }
+
+/// Creates a mock connection based on the `Validator` to `Validator` connection
+pub fn validator_mock_connection(
```

Comment:
nit: you could have all of these methods call a single private helper, e.g.,

```
fn mock_peer_and_metadata(...) -> (PeerNetworkId, ConnectionMetadata) {
    let peer = PeerNetworkId::new(network_id, peer_id);
    let metadata = mock_conn_metadata(peer, peer_role, origin, Some(protocol_ids));
    (peer, metadata)
}
```

Reply:
Yeah, I can do that.  For the couple lines I wasn't super worried